lib/k8s/resource_client.rb

```diff
@@ -224,20 +224,20 @@ def watch(labelSelector: nil, fieldSelector: nil, resourceVersion: nil, timeout:
       @transport.request(
         method: method,
         path: path,
-        response_class: K8s::API::MetaV1::List,
         query: make_query(
           'labelSelector' => selector_query(labelSelector),
           'fieldSelector' => selector_query(fieldSelector),
           'resourceVersion' => resourceVersion,
-          'watch' => '1'
+          'watch' => '1',
+          'timeoutSeconds' => timeout
```

Comment:
I suppose kube api is fine with nil here. `timeout.to_i` would make nil `0`.

Reply:
If `timeoutSeconds` is not given then Kubernetes api uses the default timeout (usually 30mins).